### PR TITLE
14.1 Kafka Handler Errors Service,  Test, Configuration

### PR DESCRIPTION
**Pull request by xalivan**

### CODE REVIEW

File: src/main/java/com/speaker/config/kafka/CustomDeserializer.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.speaker.dto.MessageDTO;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.common.errors.SerializationException;
+import org.apache.kafka.common.serialization.Deserializer;
+
+import java.util.Map;
+
+@Slf4j
+public class CustomDeserializer implements Deserializer<MessageDTO> {
```
Comment (otkalenko) [inline src/main/java/com/speaker/config/kafka/CustomDeserializer.java:14]: why do not use standart `org.springframework.kafka.support.serializer.JsonDeserializer`?

File: src/main/java/com/speaker/config/kafka/CustomSerializer.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.common.errors.SerializationException;
+import org.apache.kafka.common.header.Headers;
+import org.apache.kafka.common.serialization.Serializer;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+@Slf4j
+public class CustomSerializer implements Serializer<MessageDTO> {
```
Comment (otkalenko) [inline src/main/java/com/speaker/config/kafka/CustomSerializer.java:16]: why do not use standart `org.springframework.kafka.support.serializer.JsonSerializer`? do not see specific configuration

File: build.gradle [hunk 3]
```diff
@@ -31,6 +31,16 @@ dependencies {
     implementation group: 'io.springfox', name: 'springfox-swagger2', version: '2.9.2'
     implementation group: 'io.springfox', name: 'springfox-swagger-ui', version: '2.9.2'
 
+    implementation group: 'org.springframework.kafka', name: 'spring-kafka', version: '2.8.4'
+    testImplementation group: 'org.springframework.kafka', name: 'spring-kafka-test', version: '2.8.5'
+//    implementation group: 'org.apache.kafka', name: 'kafka-clients', version: '3.0.0'
+//    // https://mvnrepository.com/artifact/org.springframework.kafka/spring-kafka-dist
+//    implementation group: 'org.springframework.kafka', name: 'spring-kafka-dist', version: '2.8.5', ext: 'pom'
+//// https://mvnrepository.com/artifact/com.fasterxml.jackson.datatype/jackson-datatype-jsr310
+    implementation group: 'com.fasterxml.jackson.datatype', name: 'jackson-datatype-jsr310', version: '2.9.7'
```
[the remark this answers is not in the document]
Comment (xalivan) [reply build.gradle]: Yes. This is left dependence. Removed

File: src/main/java/com/speaker/service/KafkaHandlerErrorsService.java [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package com.speaker.service;
+
+public interface KafkaHandlerErrorsService<T>{
```
Comment (otkalenko) [inline src/main/java/com/speaker/service/KafkaHandlerErrorsService.java:3]: Looks like `T` must be as `BaseEntityDTO`

File: src/main/java/com/speaker/config/kafka/CustomDeserializer.java (hunk 1 above)
Comment (xalivan) [reply src/main/java/com/speaker/config/kafka/CustomDeserializer.java]: OK. This is better solution. Change made

File: src/main/java/com/speaker/config/kafka/CustomSerializer.java (hunk 2 above)
Comment (xalivan) [reply src/main/java/com/speaker/config/kafka/CustomSerializer.java]: OK. This is better solution. Change made

File: src/main/java/com/speaker/service/KafkaHandlerErrorsService.java (hunk 4 above)
Comment (xalivan) [reply src/main/java/com/speaker/service/KafkaHandlerErrorsService.java:3]: OK. Change made

**Pull request merged**